src/dfx-core/src/config/model/network_descriptor.rs

```diff
@@ -75,6 +76,26 @@ impl NetworkDescriptor {
             None => Err(NetworkMustBeLocal(self.name.clone())),
         }
     }
+
+    pub fn replica_endpoints(&self) -> Result<Vec<Url>, NetworkConfigError> {
+        self.try_into()
+    }
+}
+
+impl TryFrom<&NetworkDescriptor> for Vec<Url> {
+    type Error = NetworkConfigError;
+
+    fn try_from(network_descriptor: &NetworkDescriptor) -> Result<Self, Self::Error> {
+        network_descriptor
+            .providers
+            .iter()
+            .map(|s| {
+                Url::parse(s).map_err(|e| {
+                    NetworkConfigError::ParseProviderUrlFailed(Box::new(s.to_string()), e)
+                })
+            })
+            .collect::<Result<Self, Self::Error>>()
+    }
```

Comment:
For the sake of those of us who are mere rust mortals, can we inline this?

```suggestion
    pub fn replica_endpoints(&self) -> Result<Vec<Url>, NetworkConfigError> {
        self
            .providers
            .iter()
            .map(|s| {
                Url::parse(s).map_err(|e| {
                    NetworkConfigError::ParseProviderUrlFailed(Box::new(s.to_string()), e)
                })
            })
            .collect()
    }
}

```

Reply:
yes :D